app/instance-initializers/ember-i18n.js

```diff
@@ -14,6 +14,7 @@ export default {
     const key = 'service:i18n';
     const i18n = instance.lookup ? instance.lookup(key) : instance.container.lookup(key);
     i18n.set('locale', defaultLocale);
+    i18n.set('allowLocaleOverride', (ENV.i18n || {}).allowLocaleOverride);
```

Comment:
I think it's fine – this way the service itself is isolated, and doesn't try to pull in outside information when you unit-test it. Presumably the initializer logic could be covered by an integration test (if there was a way to stub ENV values).